### PR TITLE
literaturesuggest: added defense_date in thesis form

### LINKED ISSUE #2067: Bring back defense_date in submission form
In 1b49c9459b50fd, `defense_date` was incorrectly removed. This should be reverted.

### PR DESCRIPTION
**Pull request by Glignos**
* Added defense date field in the literature suggest thesis form (closes #2067)

### CODE REVIEW

**Comment by jmartinm** [comment]
Check commit log message styling (`git log`)

File: tests/integration/fixtures/thesis_expected.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
    "thesis_info":{
       "date":"2017-02-09T00:00:00",
+      "defense_date":"2017-02-19T00:00:00",
```
Comment (jmartinm) [inline tests/integration/fixtures/thesis_expected.json]: So this should be changed to not contain T00:00:00 after builder is amended

File: inspirehep/modules/literaturesuggest/forms.py [hunk 2]
```diff
@@ -439,6 +439,12 @@ def _is_other_language(language):
         widget=defensedate_widget,
     )
 
+    defense_date = fields.TextField(
+        label=_('Date of Submission'),
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/forms.py]: @michamos can you confirm that `defense_date` is 'Date of Submission' ?

Comment (michamos) [inline inspirehep/modules/literaturesuggest/forms.py]: no, looks like wrong copy/paste. `thesis_date` is date of submission, `defense_date` is date of defense.

Comment (Glignos) [inline inspirehep/modules/literaturesuggest/forms.py]: My bad there!I will hotfix this!

File: tests/integration/fixtures/thesis_expected.json (hunk 1 above)
Comment (Glignos) [inline tests/integration/fixtures/thesis_expected.json]: @jmartinm Yes its gonna change in my next PR

**Comment by jmartinm** [comment]
Can you check what happened with your commits? You now have two of them and the second one is mixed with a commit from @david-caro

**Comment by jmartinm** [comment]
With this change, dates look like:

screen shot 2017-03-31 at 1 40 40 pm

You can just create one combined PR together with https://github.com/inspirehep/inspire-next/pull/2155 where one commit makes all dates accept all the formats, and the other one adds defense date.

Please address https://github.com/inspirehep/inspire-next/pull/2155#issuecomment-290399426, make sure all inputs look the same, and they all accept the expected formats (YYYY-MM-DD,
YYYY-MM or YYYY )

**Comment by jmartinm** [comment]
Needs rebase to fix the 2 conflicting files.

The first commit log message should be:

```
literaturesuggest: added defense date on thesis

* Added defense date in the fields of thesis in literature suggest form. (closes #2067)
```

**Comment by jacquerie** [comment]
> ```
> literaturesuggest: added defense date on thesis
>
> * Added defense date in the fields of thesis in literature suggest form. (closes #2067)
> ```

This is not good style for three reasons:

1. Subject should use the imperative
2. Body should wrap at ~70 characters
3. It's missing the signature

File: inspirehep/modules/literaturesuggest/forms.py [hunk 3]
```diff
@@ -436,7 +424,7 @@ def _is_other_language(language):
         label=_('Date of Submission'),
         description='Format: YYYY-MM-DD, YYYY-MM or YYYY.',
         validators=[date_validator],
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/forms.py]: Note also that the commit subject is incorrect: the validator is still there, what was removed was the widget, which was completing the incomplete dates.

**Pull request closed without merging**